### DISCUSSION TITLE
Which Language?

### DISCUSSION THREAD

**Discussion by TechNightFury** [community/community#148029] [Programming Help]
### Select Topic Area

Question

### Body

I'd like to know which language is good for developing slot game engine.

**Comment by BDadmehr0** [comment]
- 1. C++
- 2. C# (unity)
- 3. Javascript/HTML5 (web-based)
- 4. Java (Android-based)
- 5. Python (Not as performant as C++ or C#)
- 6. Swift/Objective-C (for iOS)

**Comment by Stellar-1117** [comment]
C# - Ideal for using Unity for rich visuals and cross-platform compatibility.
C++ - Best for high-performance and graphics-intensive games.
JavaScript/TypeScript - Perfect for web-based slot games.
Python - Great for prototyping or backend logic.
Java - Excellent for Android or cross-platform games.

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @TechNightFury!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

**Comment by tigran-sargsyan-w** [comment]
there isn’t a single “best” language for a slot engine—pick based on your constraints. If you’re building real-money slots, your priorities are server-authoritative RNG, auditable math (RTP/volatility), and regulatory compliance; if you’re building social/casual slots, your priority is fast iteration and cross-platform visuals. 🎰
